veggieseasons/lib/screens/settings.dart

```diff
@@ -3,19 +3,199 @@
+                },
+              );
+            } else {
+              toggle = CupertinoSwitch(
+                value: false,
+                onChanged: (value) {},
+              );
+            }
+
+            items.add(SettingsItem(
+              label: veggieCategoryNames[category],
+              content: toggle,
+            ));
+          }
+
+          return ListView(
+            children: [
+              SettingsGroup(
+                items: items,
+              ),
+            ],
+          );
+        },
+      ),
+    );
+  }
+}
+
+class CalorieSettingsScreen extends StatelessWidget {
+  static const max = 1000;
+  static const min = 2600;
+  static const step = 200;
 
-class SettingsScreen extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
+    final model = ScopedModel.of<Preferences>(context, rebuildOnChange: true);
+
     return CupertinoPageScaffold(
       navigationBar: CupertinoNavigationBar(
-        middle: Text('Settings'),
+        middle: Text('Target Calorie Level'),
+        previousPageTitle: 'Settings',
       ),
       backgroundColor: Styles.scaffoldBackground,
-      child: Center(
-        child: Text('Not yet implemented.'),
+      child: ListView(
+        children: [
+          FutureBuilder<int>(
+            future: model.desiredCalories,
+            builder: (context, snapshot) {
+              final steps = <SettingsItem>[];
+
+              for (int cals = max; cals < min; cals += step) {
+                steps.add(
+                  SettingsItem(
+                    label: cals.toString(),
+                    icon: SettingsIcon(
+                      icon: Styles.checkIcon,
+                      foregroundColor: snapshot.hasData && snapshot.data == cals
+                          ? CupertinoColors.activeBlue
+                          : Styles.transparentColor,
+                      backgroundColor: Styles.transparentColor,
+                    ),
+                    onPress: snapshot.hasData
+                        ? () => model.setDesiredCalories(cals)
+                        : null,
+                  ),
+                );
+              }
+
+              return SettingsGroup(
+                items: steps,
+                header: SettingsGroupHeader('Available calorie levels'),
+                footer: SettingsGroupFooter('These are used for serving '
+                    'calculations'),
+              );
+            },
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class SettingsScreen extends StatelessWidget {
+  Widget _buildCaloriesItem(BuildContext context, Preferences prefs) {
+    return SettingsItem(
+      label: 'Target Calorie Intake',
+      icon: SettingsIcon(
+        backgroundColor: Styles.iconBlue,
+        icon: Styles.calorieIcon,
+      ),
+      content: FutureBuilder<int>(
+        future: prefs.desiredCalories,
+        builder: (context, snapshot) => Text(snapshot.data?.toString() ?? ''),
+      ),
```

Comment:
I may not visualize correctly but all table cell items that navigates needs the > chevron at the end

Reply:
Added. Should I pull the preview of the selected value as well, or include both?

Reply:
You can have both